addons/reloadlaunchers/functions/fnc_reloadLauncher.sqf

```diff
@@ -18,12 +20,27 @@
  * Public: No
  */
 
-params ["_unit","_target","_weapon","_magazine"];
-TRACE_4("params",_unit,_target,_weapon,_magazine);
+params ["_unit", "_target", "_weapon", "_magazine", "_ammo"];
+TRACE_5("params",_unit,_target,_weapon,_magazine,_ammo);
 
-_target selectWeapon _weapon;
+private _checkSelectedWeapon = isPlayer _target || {!isNull (_target getVariable ["bis_fnc_moduleRemoteControl_owner", objNull])};
 
-if (currentWeapon _target != _weapon) exitWith {};
-if (currentMagazine _target != "") exitWith {};
+// AI don't select launchers with selectWeapon/switchWeapon
+if (_checkSelectedWeapon) then {
+    _target selectWeapon _weapon;
+};
 
-_target addWeaponItem [_weapon, _magazine, true];
+// Check if the launcher has been selected (but only for players and remote controlled AI) and hadn't been reloaded
+if ((_checkSelectedWeapon && {currentWeapon _target != _weapon}) || {currentMagazine _target != ""}) exitWith {
+    // If failed, readd magazine back to reloading unit
+    [_unit, _magazine, _ammo, true] call CBA_fnc_addMagazine;
+
+    // Notify reloading unit about failure
+    [QEGVAR(common,displayTextStructured), [LLSTRING(LauncherNotLoaded)], _unit] call CBA_fnc_targetEvent;
```

Comment:
these notifications are related to https://github.com/acemod/ACE3/pull/9202
we might want to merge parts of that and add a setting?

Reply:
yeah, lets just fix it in 9202 after this pr

Reply:
@johnb432 if you want to include the changes in #9202 in this PR instead I don't mind, then I will just close that one.

Reply:
> @johnb432 if you want to include the changes in #9202 in this PR instead I don't mind, then I will just close that one.

I'll go ahead and merge #9202 into this PR.

Reply:
@Drofseh I have merged #9202.